templates/fine-tune-llm_v2/cookbooks/customize_initial_weights_and_prompt_format/README.md

```diff
@@ -0,0 +1,172 @@
+2. Customizing the chat template/ prompt format - Specify a custom prompt format for formatting input messages to easily fine-tune on _any_ data format.
+
+The Anyscale platform is uniquely suited to support both of these use-cases. This guide assumes you have familiarized yourself with the [basic fine-tuning guide](../../README.md).
+
+
+# Table of Contents
+1. [Bring your own weights](#bring-your-own-weights)
+    - [Bring models of the same architecture](#bring-models-of-the-same-architecture)
+        - [Example YAML](#example-yaml)
+        - [How do I configure access to my weights in remote storage??](#)
+            - [How do I bring my weights to Anyscale?](#how-do-I-bring-my-weights-to-Anyscale-?)
+    - [Bring checkpoints from a previous finetuning run](#bring-checkpoints-from-a-previous-finetuning-run)
+2. [Customizing the prompt format (chat template)](#customizing-the-prompt-format)
+    - [How prompt formatting works in `llmforge`](#how-prompt-formatting-works-in-llmforge)
+    - [Customizing data preprocessing and the prompt format](#customizing-data-preprocessing-and-the-prompt-format)
+
+# Bring your own weights
+
+In general, you can customize the initial weights in your fine-tuning run through two options in the YAML:
+- `initial_base_model_ckpt_path` : Path to the base model weights you wish to start with
+- `initial_adapter_model_ckpt_path`: Path to the adapter (LoRA) weights you wish to start with
+
+## Bring models of the same architecture
+You can fine-tune a model similar in architecture to the Llama or Mistral family of models to fine-tune on the Anyscale Platform. For example, to fine-tune Llama Guard 2, you can 
+specify the model ID and `initial_base_model_ckpt_path` as below:
+### Example YAML
+
+```yaml
+model_id: meta-llama/Meta-Llama-3-8B
+initial_base_model_ckpt_path: s3://my-bucket/llama-guard-2
+train_path: s3://air-example-data/gsm8k/train.jsonl
+valid_path: s3://air-example-data/gsm8k/test.jsonl
+...
+```
+
+
+The overarching idea is that specifying a model ID will provide context to the architecture of the LLM, which will guide how the model will be further trained. It won't necessarily mean that the model that is specified here is the model that will be fine-tuned. For that, we will rely on the weights that are provided. Knowing that Llama 3 8B model (or any other Llama model for that matter) shares the same architecture as the Llama Guard 2 model makes it a suitable choice for the model ID. However, note that this would still use the same chat-templating / prompt formatting as Llama-3 while starting to fine-tune with Llama Guard 2 weights. For the specific case of Llama Guard 2, we need customization even in the prompt format which will be outlined below. 
```

Comment:
this is very confusing (that model_id gives architecture hint) - isn't the output artifact also stored under model_id:....? Can we instead add some config like base_model_id

Reply:
yes the `model_id` will be used for inference. `base_model_id` is an option, but the root case for the confusion is that we don't allow model ID beyond Llama and Mistral family. So the fix is going to be opening that up and allowing for other model ids (So for this example, it would have been nice to just say `model_id: meta-llama/Meta-Llama-Guard-2-8B` ). 

cc @kouroshHakha for more thoughts on this.

Reply:
what's the ETA to fix that? I think it would be better to hold back this cookbook until that is fixed